## 01.One sample t test 2 sided

In [1]:
import pandas as pd
import numpy as np

In [2]:
df_production=pd.read_csv(r'D:\Imarticus Database\Hypothesis Testing Data Set\Production.csv')

* Two tail que

Que.1 The voilent industry furniture for home and office
Muee=Mean=The average number of furniture produced by  Machine A

Hypotesis
H0: Mean of items Produced by Machine A is 10,i.e. μ=30
    Against
H1: Mean of items Produced by Machine A is not equal to 10 i.e. μ!=30 

In [3]:
from scipy.stats import ttest_1samp

In [30]:
#Conduct T-test
ttest,pvalue=ttest_1samp(df_production["Items Produced"],30)

In [32]:
print(ttest,pvalue)

1.2674612253287467 0.23680061493494728


In [33]:
if pvalue>0.05:
    print("Accept H0 ")
else:
    print("Reject H0")

Accept H0 


* we can see Pvalue(0.236)>Significace(0.05) hence we accept H0
 from here we can conclude that μ = 30

* One tail que

* Que. bulb life
Hypotesis
Ho: μ=45000
H1: μ<=45000

In [34]:
# Load Data
df_lifeSpan= pd.read_csv(r'D:\Imarticus Database\Hypothesis Testing Data Set\LifeSpan.csv')
df_lifeSpan

life
0   35000
1   49000
2   44000
3   45900
4   43400
5   50000
6   47200
7   32700
8   33200
9   43000
10  46300
11  47800
12  32100
13  33200
14  33900

In [7]:
# Assume first is two tail test
t_test_less = ttest_1samp(df_lifeSpan['life'],45000)

In [8]:
# so divide Two tail by 2 to get One Tail
p_value_less=t_test_less.pvalue/2

In [9]:
p_value_less

0.022628491566062506

In [10]:
# Since P-Value(0.022)<0.05 (Significance) hence we reject H0  μ<=45000

## Two Sample T test

* Que. X: Diet A weight
       y: Diet B weight 
μA=Average weight of A
μB=Average weight of B

H0:μA=μB (Average weight of Both is Same)
H1:μA!=μB(Average weight of Both is Different)


In [11]:
df_diet=pd.read_csv(r'D:\Imarticus Database\Hypothesis Testing Data Set\Diet_data.csv')

In [12]:
#T-Test for Eqaulity of Means
from scipy.stats import ttest_ind

In [13]:
ttest_ind(df_diet["Diet A"],df_diet["Diet B"])
# Pvalue> alpha i.e.0.05 so we Fail to Reject H0
# so Mean are  equal

Ttest_indResult(statistic=0.5169140149580205, pvalue=0.6115094596012607)

## Paird T-test

x=Hours of Sleep before Meditation
y= after 
μx:Average hours of Sleep before Meditation
μy:Average after
H0: μx=μy
H1: μx!=μy

In [24]:
df_sleepdata=pd.read_csv(r'D:\Imarticus Database\Hypothesis Testing Data Set\Sleep_data.csv')
df_sleepdata

Hours of Sleep (Before)  Hours of Sleep (After)
0                      5.0                     7.0
1                      6.0                     7.5
2                      6.5                     9.0
3                      5.5                     7.0
4                      6.5                     8.0
5                      7.0                     7.5
6                      8.0                     7.0
7                      7.5                     7.0
8                      6.0                     7.0
9                      7.0                     6.5

In [15]:
from scipy.stats import ttest_rel

In [23]:
ttest_rel(df_sleepdata['Hours of Sleep (Before)'],df_sleepdata['Hours of Sleep (After)'])

Ttest_relResult(statistic=-2.2785119632047284, pvalue=0.048680802344322635)

 Pvalue(0.048)<0.05 so that we are REJECTING H0
 we can conclude that average Hours of Sleep after Sleep is not same before and after meditation 

# Z-test

* Que. 

A random Sample of 40 glass rods is taken from a lot manufacturing under a new process. They are tested for their breaking strength.Can we claim that the breaking strength is equal to 55lbs?

X: Breaking Strength of Glass rod
thus μ= Average breaking stength of Glass
H0: μ =55lbs(Average breaking stength of Glass=55 lbs)
H1: μ!= 55 lbs(Average breaking stength of Glass is NOT 55 lbs)

In [42]:
# Load Data
df_strength= pd.read_csv(r'D:\Imarticus Database\Hypothesis Testing Data Set\Strength.csv')

In [43]:
#Import Libraries
from statsmodels.stats import weightstats as stests

In [39]:
# Conduct Test
stests.ztest(df_strength["Strength(in lbs)"],value=55)

(-4.431740748679182, 9.347536680921983e-06)

Since Pvalue(0)<Significance(0.05),Hence we Reject Null Hypotesis
Hence glassrod Strength is not 55 lbs

### Two Sample Test(Z)

Que. Two types of fertilizers are teasted on farm lands of the same size.The yields in quintals are given below.Can we conclude that two fertilizers have Same Yield?

X: Fertilizer 1
Y: Fertilizer 2
ua=Average Yield due to Fertilizer 1
ub=Average Yield due to Fertilizer 2
H0: Both Yield Same (ua=ub)
Ha: Both Yield Different(ua!=ub)

In [ ]:
# Load Data
df_fertilizers=pd.read_csv(r'D:\Imarticus Database\Hypothesis Testing Data Set\Fertilizer.csv')
#Import Libraries
from statsmodels.stats import weightstats as stests

In [50]:
# Conduct Ztest
stests.ztest(df_fertilizers["Fertilizer 1"],df_fertilizers["Fertilizer 2"])

(-1.1390306751872912, 0.25469036008906554)

Since Pvalue(0.2546)>0.05 so that H0 is ACCEPTED, Both yield SAME

# Shapiro-Wilk Test

In [2]:
import numpy as np

In [4]:
# Create data random Sample
rain_in_mm=np.random.random(200)*100

H0: Data is Normally Distributed
Ha: Data is NOT Normally Distributed

In [6]:
# Import shapiro from scipy library
from scipy.stats import shapiro
# conduct Normality Test
stat,p= shapiro(rain_in_mm)

print("Test Statictics =",stat,"and Pvalue =",p)

Test Statictics = 0.9536012411117554 and Pvalue = 4.263997197995195e-06


Since Pvalue(4.46)<0.05 so that H0 is Rejected 
Hence we Can Conclude that Data is NOT Normally Distributed

 # Jarque-Bera test

In [15]:
# create the data by generating generating a random sample
diameter_in_mm = np.random.normal(loc=4,scale=0.7,size=6000)  \
                     + np.random.normal(loc=5,scale=0.1,size=6000)

 H0: The Data is Normally Distributed
 Against
H1: The Data is NOT Normally Distributed

In [16]:
# Import jarque_bera from the scipy library
from scipy.stats import jarque_bera

# conduct the Normality Test
stat,p=jarque_bera(diameter_in_mm)

print("The Test Statictics is ",stat,"and its Pvalue is",p)

The Test Statictics is  2.17367077340441 and its Pvalue is 0.33728217423766615


Since Pvalue(0.337)>0.05,Hence we Accept H0.
We may conclude that Data is Normally Distributed